table.go

```diff
@@ -728,6 +745,55 @@ func (table Table[T]) DeleteEntityWithID(ctx context.Context, entityID string) (
 	return entity, table.DeleteEntity(ctx, entity)
 }
 
+// DeleteEntityVersionWithID deletes the specified version of the entity from the table.
+func (table Table[T]) DeleteEntityVersionWithID(ctx context.Context, entityID, versionID string) (T, error) {
```

Comment:
I wonder if the name of this function is a little misleading. It will delete if it's the only version, but if it's one in the middle, it doesn't really delete it, it just creates a new version based on the previous one. It's more like "rollback to version" or something like that.
It will effectively look like you deleted all the versions back to a specific version, but in some instances it will not be removing them. I could see this being an issue if you were trying to clear the DB of sensitive information or something.

Reply:
Hmm... I'm not sure I follow. The `DeleteItem` call always deletes the specified version at the end, unless the entire entity was deleted because it was the only remaining version. If the version being deleted is the current version, then the `UpdateEntityVersion` call updates the indexes to the prior version. In that sense, it's a roll-back of one version. If you're deleting a version that is not the current version, then there's no impact on the indexes. It just deletes the specified item in the entity row.